x/twap/store_test.go

```diff
@@ -189,3 +188,173 @@ func (s *TestSuite) TestGetRecordAtOrBeforeTime() {
 		})
 	}
 }
+
+// TestPruneRecordsBeforeTime tests that all twap records earlier than
+// current block time - given time are pruned from the store.
+func (s *TestSuite) TestPruneRecordsBeforeTime() {
+	tMin2Record, tMin1Record, baseRecord, tPlus1Record := s.createTestRecordsFromTime(baseTime)
+
+	// non-ascending inserton order.
+	allTestRecords := []types.TwapRecord{tPlus1Record, tMin1Record, baseRecord, tMin2Record}
+
+	tests := map[string]struct {
+		recordsToPreSet []types.TwapRecord
+
+		beforeTime time.Time
+
+		expectedKeptRecords []types.TwapRecord
+
+		expErr bool
+	}{
+		"base time, 1 record before base time (deleted)": {
+			recordsToPreSet: []types.TwapRecord{tMin1Record},
+
+			beforeTime: baseTime,
+
+			expectedKeptRecords: []types.TwapRecord{},
+		},
+		"base time, 2 records before base time (both deleted)": {
+			recordsToPreSet: []types.TwapRecord{tMin1Record, tMin2Record},
+
+			beforeTime: baseTime,
+
+			expectedKeptRecords: []types.TwapRecord{},
+		},
+		"base time, 1 record at base time (not deleted)": {
+			recordsToPreSet: []types.TwapRecord{baseRecord},
+
+			beforeTime: baseTime,
+
+			expectedKeptRecords: []types.TwapRecord{baseRecord},
+		},
+		"base time, 1 record after base time (not deleted)": {
+			recordsToPreSet: []types.TwapRecord{tPlus1Record},
+
+			beforeTime: baseTime,
+
+			expectedKeptRecords: []types.TwapRecord{tPlus1Record},
+		},
+		"base time minus 1, 2 records before (deleted), 1 records at (deleted), 1 records after (not deleted)": {
+			recordsToPreSet: allTestRecords,
+
+			beforeTime: tMin1Record.Time,
+
+			expectedKeptRecords: []types.TwapRecord{tMin1Record, baseRecord, tPlus1Record},
+		},
+		"base time minus 2 - 0 records before - all kept": {
+			recordsToPreSet: allTestRecords,
+
+			beforeTime: tMin2Record.Time,
+
+			expectedKeptRecords: []types.TwapRecord{tMin2Record, tMin1Record, baseRecord, tPlus1Record},
+		},
+		"base time plus 2 - all records before - all deleted": {
+			recordsToPreSet: allTestRecords,
+
+			beforeTime: tPlus1Record.Time.Add(time.Second),
+
+			expectedKeptRecords: []types.TwapRecord{},
+		},
+		"no pre-set records - no error": {
+			recordsToPreSet: []types.TwapRecord{},
+
+			beforeTime: baseTime,
+
+			expectedKeptRecords: []types.TwapRecord{},
+		},
+	}
+	for name, tc := range tests {
+		s.Run(name, func() {
+			s.SetupTest()
+			s.preSetRecords(tc.recordsToPreSet)
+
+			ctx := s.Ctx
+			twapKeeper := s.twapkeeper
+
+			err := twapKeeper.PruneRecordsBeforeTime(ctx, tc.beforeTime)
+			if tc.expErr {
+				s.Require().Error(err)
+				return
+			}
+			s.Require().NoError(err)
+
+			s.validateExpectedRecords(tc.expectedKeptRecords)
+		})
+	}
+}
+
+func (s *TestSuite) TestGetAllHistoricalTimeIndexedTWAPs() {
```

Comment:
I think we can combine what we're testing with the test above by calling `GetAllHistoricalPoolIndexedTWAPs` before we prune in the test above

Reply:
I see it would. However, each test should be focused on testing one thing only, we shouldn't be testing multiple units of computation in one fixture